File: src/log.c
/* NyxNode
 * Author: Jérôme ODIER <jerome.odier@lpsc.in2p3.fr>
 * SPDX-License-Identifier: GPL-2.0-only (Mongoose backend) or GPL-3.0+
 */

/*--------------------------------------------------------------------------------------------------------------------*/

#include "nyx_node_internal.h"

/*--------------------------------------------------------------------------------------------------------------------*/

nyx_log_level_t nyx_log_level = NYX_LOG_LEVEL_INFO;

/*--------------------------------------------------------------------------------------------------------------------*/

STR_t nyx_log_level_to_str(nyx_log_level_t level)
{
    switch(level)
    {
        case NYX_LOG_LEVEL_NONE:
            return "NONE";
        case NYX_LOG_LEVEL_FATAL:
            return "\033[91;1mFATAL\033[0m";
        case NYX_LOG_LEVEL_ERROR:
            return "\033[91;1mERROR\033[0m";
        case NYX_LOG_LEVEL_WARN:
            return "\033[93;1mWARN\033[0m";
        case NYX_LOG_LEVEL_INFO:
            return "\033[32mINFO\033[0m";
        case NYX_LOG_LEVEL_DEBUG:
            return "\033[34mDEBUG\033[0m";
        case NYX_LOG_LEVEL_TRACE:
            return "\033[36mTRACE\033[0m";
    }

    return "???";
}

/*--------------------------------------------------------------------------------------------------------------------*/

void nyx_set_log_level(nyx_log_level_t level)
{
    nyx_log_level = level;
}

/*--------------------------------------------------------------------------------------------------------------------*/

File: nyx_node/log.py
# -*- coding: utf-8 -*-
########################################################################################################################
# NyxNode
# Author: Jérôme ODIER <jerome.odier@lpsc.in2p3.fr>
# SPDX-License-Identifier: GPL-3.0+
########################################################################################################################

from . import bind

########################################################################################################################

NyxLogLevel = bind.NyxLogLevel

########################################################################################################################

def nyx_set_log_level(level: bind.NyxLogLevel) -> None:
    """! @brief Sets the log level threshold.

    @param level Log level threshold.
    """

    if not isinstance(level, NyxLogLevel):

        raise TypeError('Expected NyxLogLevel enum')

    bind.lib.nyx_set_log_level(level)

########################################################################################################################

__all__ = [name for name in globals() if name.lower().startswith('nyx')]

########################################################################################################################
